docs/features.rst

```diff
@@ -23,66 +22,106 @@ For now, we extract the features for the whole recordings, and store them in sep
-Currently, these arguments are the sum of all possible arguments for all feature types - we will likely refactor that
-in the future, and we might either move this section into a separate manifest or drop it entirely and store
-only the non-default values in each feature document.
-
-The feature documents contain the information necessary to tie them to a particular recording - ``start``, ``duration``,
+    dither: 0.0
+    energy_floor: 0.1
+    frame_length: 0.025
+    frame_shift: 0.01
+    min_duration: 0.0
+    preemphasis_coefficient: 0.97
+    raw_energy: true
+    remove_dc_offset: true
+    round_to_power_of_two: true
+    window_type: povey
+    type: spectrogram
+
+And the corresponding configuration class:
+
+.. autoclass:: lhotse.features.SpectrogramConfig
+  :members:
+
+The feature matrices manifest is a list of documents.
+These documents contain the information necessary to tie the features to a particular recording: ``start``, ``duration``,
 ``channel`` and ``recording_id``. They currently do not have their own IDs.
 They also provide some useful information, such as the type of features, number of frames and feature dimension.
 Finally, they specify how the feature matrix is stored with ``storage_type`` (currently ``numpy`` or ``lilcom``),
 and where to find it with the ``storage_path``. In the future there might be more storage types.
 
+.. code-block:: yaml
+
+    - channel_id: 0
+      duration: 16.04
+      num_features: 23
+      num_frames: 1604
+      recording_id: recording-1
+      start: 0.0
+      storage_path: test/fixtures/libri/storage/dc2e0952-f2f8-423c-9b8c-f5481652ee1d.llc
+      storage_type: lilcom
+      type: fbank
+
+
+Creating custom feature extractor
+*********************************
+
+There are two components needed to implement a custom feature extractor: a configuration and the extractor itself.
+We expect the configuration class to be a dataclass, so that it can be automatically mapped to dict and serialized.
+The feature extractor should inherit from :ref:`lhotse.features.FeatureExtractor`,
+and implement a small number of methods/properties.
+The base class takes care of initialization (you need to pass a config object), serialization to YAML, etc.
+A minimal, complete example of adding a new feature extractor:
+
+.. code-block::
+
+    @dataclass
+    class ExampleFeatureExtractorConfig:
+        frame_shift: Seconds = 0.01
+
+    class ExampleFeatureExtractor(FeatureExtractor):
+        name = 'example-feature-extractor'
+        config_type = ExampleFeatureExtractorConfig
+
+        def extract(self, samples: np.ndarray, sampling_rate: int) -> np.ndarray:
+            f, t, Zxx = stft(samples, sampling_rate, noverlap=round(self.frame_shift * sampling_rate))
+            return np.abs(Zxx)
+
+        @property
+        def frame_shift(self) -> Seconds:
+            return self.config.frame_shift
```

Comment:
It might be a good idea to mention at some point that the lilcom compression method uses a fixed precision that doesn't depend on the magnitude of the thing being compressed, so it's better suited to log-energy features than energy features. (Of course it can be disabled).

Reply:
Added to docs.